substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/util/TopologicalStyleProvider.java

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.substationdiagram.util;
+
+import static com.powsybl.substationdiagram.svg.SubstationDiagramStyles.BUS_STYLE_CLASS;
+import static com.powsybl.substationdiagram.svg.SubstationDiagramStyles.GRID_STYLE_CLASS;
+import static com.powsybl.substationdiagram.svg.SubstationDiagramStyles.LABEL_STYLE_CLASS;
+import static com.powsybl.substationdiagram.svg.SubstationDiagramStyles.SUBSTATION_STYLE_CLASS;
+import static com.powsybl.substationdiagram.svg.SubstationDiagramStyles.WIRE_STYLE_CLASS;
+import static com.powsybl.substationdiagram.svg.SubstationDiagramStyles.escapeClassName;
+import static com.powsybl.substationdiagram.svg.SubstationDiagramStyles.escapeId;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import com.powsybl.basevoltage.BaseVoltageColor;
+import com.powsybl.iidm.network.Branch.Side;
+import com.powsybl.iidm.network.BusbarSection;
+import com.powsybl.iidm.network.DanglingLine;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Line;
+import com.powsybl.iidm.network.Load;
+import com.powsybl.iidm.network.ShuntCompensator;
+import com.powsybl.iidm.network.StaticVarCompensator;
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+import com.powsybl.iidm.network.TopologyVisitor;
+import com.powsybl.iidm.network.TwoWindingsTransformer;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.substationdiagram.library.ComponentType;
+import com.powsybl.substationdiagram.model.Edge;
+import com.powsybl.substationdiagram.model.Graph;
+import com.powsybl.substationdiagram.model.Node;
+import com.powsybl.substationdiagram.model.Node.NodeType;
+import com.powsybl.substationdiagram.svg.DefaultSubstationDiagramStyleProvider;
+import com.powsybl.substationdiagram.svg.SubstationDiagramStyles;
+
+import javafx.scene.paint.Color;
```

Comment:
SVG generation is intended to be execute on server side and we not should be dependent on JavaFX on server side.
Could you replace javafx.scene.paint.Color by something else that does not depend on JavaFX ou Swing or AWT.
When we will switch to Java 9 module system, we won't add Java dependency on server side.

Reply:
It's possible to compute  color gradient using only RGB values, but the stat and end colors must be specified.  I could add minColor and maxColor parameters to BaseVoltageColor configuration:

```
baseVoltages:
    - name: "225"
      minValue: 180
      maxValue: 300
      color: "#00FF00"
      minColor: "#49FA6C"
      maxColor: "#226304"
      profile: "TEST"
```

It is ok?

Reply:
minColor and maxColor has nothing to do in base voltage config.
Why not a "delta" in config specific to substation diagram.
How your comment is related to JavaFX color import ?

Reply:
As we cannot use JavaFX or AWT or Swing Color classse, I tried to work only with RGB codes. I do not know how to get different variants of a given color ensuring that it remains in the same color group ( N red color variants for example). Instead if we configure the min and max color as a ligh red and a dark red it is possible to compute N color variants in that range.

Reply:
Did you search for a third party library non dependent to JavaFX which would have this kind of color scale generation?
Maybe we could try to port some javascript code in Java like this one: https://hihayk.github.io/scale/ ?